### ISSUE TITLE
Key - Formatting lint ( lowerCamelCase / PascalCase / snake_case)

### ISSUE THREAD

**Issue by rufreakde** [adrienverge/yamllint#404]
# Question
Is it possible to lint .yaml files for `key` formatting if you want to enforce a specific key format or maybe only allow 1 or 2 formats instead of allowing everything? Or maybe a Regex option to check keys so users can provide their own regex check for keys.

**Comment by adrienverge** [comment]
Hello, is this question the same as https://github.com/adrienverge/yamllint/issues/392?

**Comment by rufreakde** [comment]
@adrienverge it is similar but not the same.

A linter is also checking styleguides for structures your team defines.
Most linters have a [variable-name-rule](https://palantir.github.io/tslint/rules/variable-name/) or another example [here](https://eslint.org/docs/rules/camelcase).

Since for yaml we do not want to **validate any "content/values"** it would not make sense as mentioned in the #392.

But it would make sense for a consistent behaviour **to validate "keys"** which correspond to variables in "typical" linters.
What do you think?

**Comment by adrienverge** [comment]
If I understand your point, #392 is about forcing *values* to match a regex, while your issue is about forcing *keys* to match a regex. Right?

If this is the case, the same answer as for #392 applies here.

> A linter is also checking styleguides for structures your team defines.

Maybe some linters took the liberty to do more than linting, but yamllint doesn't. To me, custom requirements your team defines should be handled by a custom script.

**Comment by rufreakde** [comment]
Okay thanks @adrienverge. I Understand that yamllint was not designed for this kind of purpose.

One last question could you elaborate why `key` pattern matching is not validating the form I mean VAR_OK var_ok VarOk and varOK is also form in that sense.
for example:
line too long
missing document start "---"
too many spaces inside braces

Are rules that apply best practices for form of yaml documents - but I am not understanding why consistency in naming `key`'s is not best practice. It might be very helpful if you work with 3rd party tools like helm and kubernetes it would make static checking much simpler to have it in one linter tool.

Thank you for taking your time to answer my questions 👍

**Comment by adrienverge** [comment]
Sure. The aim of yamllint is to provide a set of rules to validate the *form* of a YAML file. These rules try to be as general-purpose as possible, and most of them stem from the official YAML specification and UNIX good practices. As a result, they often relate to YAML punctuation, special characters, and other forms of formatting that don't affect the value contained by the YAML represensation (i.e. the value if converted to JSON).

In summary, yamllint rules are about the *form*, not the *content*.

**Comment by rufreakde** [comment]
Makes sense its a tool for validity of YAML in general and not a tool to have consistentcy for YAMLs.

I paste the [spec ](https://yaml.org/) destination for others here thanks again! :)
